docs/src/extend/custom-rules.md

```diff
@@ -635,73 +688,42 @@ module.exports = {
-var nodeSourceWithFollowing = sourceCode.getText(node, 0, 2);
-```
-
-In this way, you can look for patterns in the JavaScript text itself when the AST isn't providing the appropriate data (such as location of commas, semicolons, parentheses, etc.).
-
-### Accessing Comments
-
-While comments are not technically part of the AST, ESLint provides a few ways for rules to access them:
-
-#### sourceCode.getAllComments()
-
-This method returns an array of all the comments found in the program. This is useful for rules that need to check all comments regardless of location.
-
-#### sourceCode.getCommentsBefore(), sourceCode.getCommentsAfter(), and sourceCode.getCommentsInside()
-
-These methods return an array of comments that appear directly before, directly after, and inside nodes, respectively. They are useful for rules that need to check comments in relation to a given node or token.
-
-Keep in mind that the results of this method are calculated on demand.
-
-#### Token traversal methods
-
-Finally, comments can be accessed through many of `sourceCode`'s methods using the `includeComments` option.
+**Note:** If your rule schema uses JSON schema [`$ref`](https://json-schema.org/understanding-json-schema/structuring.html#ref) properties, you must use the full JSON Schema object rather than the array of positional property schemas. This is because ESLint transforms the array shorthand into a single schema without updating references that makes them incorrect (they are ignored).
 
 ### Accessing Shebangs
 
-Shebangs are represented by tokens of type `"Shebang"`. They are treated as comments and can be accessed by the methods outlined above.
+[Shebangs (#!)](https://en.wikipedia.org/wiki/Shebang_(Unix)) are represented by the unique tokens of type `"Shebang"`. They are treated as comments and can be accessed by the methods outlined in the [Accessing Comments](#accessing-comments) section, such as `sourceCode.getAllComments()`.
 
 ### Accessing Code Paths
 
-ESLint analyzes code paths while traversing AST.
-You can access that code path objects with five events related to code paths.
+ESLint analyzes code paths while traversing AST. You can access code path objects with five events related to code paths. For more information, refer to [Code Path Analysis](code-path-analysis).
+
+### Deprecated `context` Methods
 
-[details here](code-path-analysis)
+Please note that the following `context` methods have been deprecated and will be removed in a future version of ESLint:
+
+* `getComments()`: Replaced by `SourceCode#getCommentsBefore()`, `SourceCode#getCommentsAfter()`, and `SourceCode#getCommentsInside()`.
+* `getTokenOrCommentBefore()`: Replaced by `SourceCode#getTokenBefore()` with the `{ includeComments: true }` option.
+* `getTokenOrCommentAfter()`: Replaced by `SourceCode#getTokenAfter()` with the `{ includeComments: true }` option.
+* `isSpaceBetweenTokens()`: Replaced by `SourceCode#isSpaceBetween()`
+* `getJSDocComment()`
```

Comment:
These are deprecated `SourceCode` methods.

Reply:
it says that in the first part of the section and the title. do you want any further documentation here?

Reply:
It says they're deprecated `context` methods, but these are `SourceCode` methods, not `context` methods.

Reply:
fixing